packages/scenario/test/xpath/functions/base64-decode.test.ts

```diff
@@ -0,0 +1,278 @@
+ *    (first?) porting push exercise other XPath functionality. And while XPath
+ *    functionality has by far the strongest coverage up to this point, a few
+ *    nuanced differences with JavaRosa have already been picked up in a handful
+ *    of ported tests that also exercise XPath functionality, and I expect we'd
+ *    benefit overall by sharing a great deal more XPath test coverage with
+ *    JavaRosa across the board.
+ *
+ * 2. It'll be easier to track and discuss this if it follows the same general
+ *    porting strategy as everything else.
+ *
+ * 3. It would be rather easier to miss the known failures indefinitely there,
+ *    when we're otherwise tracking them in `scenario`.
+ *
+ * My hunch is that the best first step is to file an issue for feature support,
+ * and defer any decision on where/how to test it until implementation.
+ *
+ * Other upfront observations:
+ *
+ * - It was initially surprising that we didn't have tests for this function
+ *   already. It appears to be relatively new to the spec, introduced shortly
+ *   after the web forms `xpath` package was initially added.
+ *
+ * - Given we don't currently implement the function, it's expected that all of
+ *   the ported tests will initially fail pending implementation.
+ *
+ * - I was confused by its placement in the spec under
+ *   {@link https://getodk.github.io/xforms-spec/#fn:Geographic-Functions | Geographic Functions}.
+ *   It seems likely this was unintended, and the function was just added "at
+ *   the end". There are a handful of other functions in the spec document that
+ *   seem to have an odd place in the information hierarchy.
+ *
+ * - I'd like to propose **at least** a pass to find better placement for this
+ *   and those handful of other functions. I'd also like to potentially
+ *   reconsider the table-with-headings structure of the XPath Functions
+ *   catalogue overall: besides being prone to the observed organizational
+ *   confusion, the markup semantics seem poorly structured for accessibility,
+ *   and I've generally found it fussy for navigation.
+ *
+ * - The first five tests feel well suited for a parameterized/table test.
+ *
+ * Slightly after upfront:
+ *
+ * - Evidently the first five tests were originally parameterized, and that was
+ *   relaxed to a shared setup function in review. Something to keep in mind for
+ *   prospects of sharing with JavaRosa.
+ *
+ * Overall, for consideration when we implement the feature/decide how to
+ * approach its testing:
+ *
+ * - It's concerning that all of the tests are concerned with text data. It's
+ *   good that there's a range of unicode tests, but it's notable that at least
+ *   one has been noted for its curious behavior with the unicode input text
+ *   producing "garbage" code points. I'd like to consider tests exercising
+ *   non-text input, for instance some of the binary data associated with
+ *   formats we presently support in itext media.
```

Comment:
The current description is that the encoded value is expected to represent a UTF-8 encoded string. What would the use case be for another kind of encoded value and how could it be used?

Reply:
Yeah, I see that now. I think I was moving really quickly and my brain autofilled some blanks without really looking to closely at the spec (other than a quick, sort of anxious look thinking I'd somehow missed it in the original `xpath` pass).

My first thought was that I'd use it for embedding arbitrary form attachments, sidestepping anything to do with `jr:` URLs, uploads e.g. in Central, etc. Roughly analogous to using `data:` URLs across a wide variety of web technologies. This was my frame of reference, because they're commonly base64 encoded (and combined with a MIME type).